crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -225,6 +376,9 @@ pub struct ParamSet<'w, 's, T: SystemParam> {
 /// The [`SystemParamState`] of [`ParamSet<T::Item>`].
 pub struct ParamSetState<T: for<'w, 's> SystemParamFetch<'w, 's>>(T);
 
+/// The [`SystemParamState`] of [`ParamSet<T::Item>`].
+pub struct OptionParamSetState<T: for<'w, 's> SystemParamFetch<'w, 's>>(T);
```

Comment:
What is the purpose of this type?

Reply:
Oh, my bad. That was me experimenting with the idea of optional param sets. I decided it wasn't necessary, because you could have a set of optional system params. But I guess I forgot to remove this struct. Thanks for the catch. Clearly I need to go over these changes.